src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java

```diff
@@ -114,10 +134,18 @@ public HarvestServiceImpl(HttpHarvester harvester,
       throw new ServiceException("Error harvesting records ", e);
     }
 
+    if(records.size() == maxRecords){
+      hasReachedRecordLimit.set(true);
+    }
+
     if (records.isEmpty()) {
       throw new ServiceException("Provided file does not contain any records", null);
     }
-    return records;
+    return new HarvestContent(hasReachedRecordLimit,records);
+  }
+
+  protected void setMaxRecords(int limitRecords){
```

Comment:
I guess this is for tests.
Since this is a necessity for the class now, if you want you could inject the value as a constructor parameter @Value("${sandbox.dataset.max-size}"). In that way the setter is not required and the constructor can initialize this value in the tests. 
Although on each change on the limit(which could be in fact more secure) you'd have to create a new object